internal/watcher/watcher.go

```diff
@@ -215,7 +215,7 @@ func (w *Watcher) watch(ctx context.Context, dirs []string, initError chan<- err
 	for {
 		select {
 		case event, ok := <-fsWatcher.Events:
-			if !ok {
+			if !ok || event.Name == "" {
```

Comment:
On test cases that rename the file, we get an empty `rename` event after creating the new file (really not sure why, could be a `fsnotify` bug). Then, on cases where we rename and update the file, we also get an empty `write` event as well...